### PR TITLE
Add a workflow for deploying to gh-pages branch

### PR DESCRIPTION
**Pull request by meghanrjones**
This PR adds a workflow for deploying the built pages to the gh-pages branch, based on the workflows in the PyGMT, GMT, and website repositories.

I haven't set up GitHub pages to work from a branch before (previously used a separate repo for the built files), so feedback would be greatly appreciated. In addition, I have a couple specific questions:

- Do I need to setup an orphan gh-pages branch or would `@actions/checkout` work for creating this?
- Currently, I have set up the workflow to place the built docs in an `oceania` directory in the `gh-pages` branch if a release is made and a `test` directory in `gh-pages` for non-releases. Does anyone have better ideas for how to manage documentation associated with staged changes in `test.generic-mapping-tools.org` versus published changes in `oceania.generic-mapping-tools.org`? One option would be to only publish to the `oceania` directory if there's a change to `doc/changes.rst`, though I do not know the best way to set that up.

Any thoughts or feedback, @GenericMappingTools/gmt-maintainers and @weiji14?

Part of GenericMappingTools/gmt#5867.

### CODE REVIEW

File: .github/docs.yml
Comment (seisman) [inline .github/docs.yml:52]: No need to check `matrix.os` here.
```suggestion
        if: github.event_name == 'push'
```

**Comment by seisman** [comment]
> * Currently, I have set up the workflow to place the built docs in an `oceania` directory in the `gh-pages` branch if a release is made and a `test` directory in `gh-pages` for non-releases.

It's unclear to me why you want to have two separate directories. I thought that the `test` domain is only used for testing purpose and if the testing data work as expected, then they will be moved into the `oceanic` domain. If there are any documentation changes in the `test` domain, the new documentation should be in a PR (using Vercel), not being deployed to gh-pages branch.

**Comment by PaulWessel** [comment]
We want to decouple data releases from GMT releases.  Right now, I plan to add several more data sets but there is no reason that has to coincide with GMT releases.  But with the data documentation in the GMT documentation it is a problem, hence the split.

**Comment by meghanrjones** [comment]
> > * Currently, I have set up the workflow to place the built docs in an `oceania` directory in the `gh-pages` branch if a release is made and a `test` directory in `gh-pages` for non-releases.
> 
> It's unclear to me why you want to have two separate directories. I thought that the `test` domain is only used for testing purpose and if the testing data work as expected, then they will be moved into the `oceanic` domain. If there are any documentation changes in the `test` domain, the new documentation should be in a PR (using Vercel), not being deployed to gh-pages branch.

OK, I am fine with using the PRs as a staging ground for the documentation, similar to the `test` domain being the staging ground for the data. We can simply not merge the PRs until the data are migrated to `oceania`. I will adjust the workflow.

**Comment by meghanrjones** [comment]
In order to remove the original docs from the GMT repository before 6.3, I will manually setup the gh-pages branch for now and come back to this PR after the release.

**Comment by meghanrjones** [comment]
I revised the workflow according to the [build workflow for the website](https://github.com/GenericMappingTools/website/blob/master/.github/workflows/build.yml) with the exception that the docs are deployed to the default `gh-pages` branch in the same repository as the code using the appropriate token.

@GenericMappingTools/gmt-maintainers, would anyone be willing to review this workflow to prepare for the addition of the new datasets?

**Review by seisman — APPROVED** [review]
Looks good to me.

**Pull request merged**
